service_catalog/tasks.py

```diff
@@ -16,15 +17,15 @@ def sync_tower(tower_id, job_template_id=None):
         logger.info(f"[sync_tower] sync tower server with id: {tower_id}")
     else:
         logger.info(f"[sync_tower] sync one job template({job_template_id}) in tower server with id: {tower_id}")
-    tower_server = TowerServer.objects.get(id=tower_id)
+    tower_server = get_object_or_404(TowerServer, id=tower_id)
```

Comment:
Rollback
We are in the worker. raising a 404 will not be catch by the controller.
And this zone is safe. Objects are sent by the controller.